File: src/test/java/com/ddd/trainings/idea/domain/TrainingDurationVOTest.java
package com.ddd.trainings.idea.domain;

import static org.assertj.core.api.Assertions.assertThat;

import com.ddd.trainings.TrainingTests;
import org.junit.jupiter.api.Test;

class TrainingDurationVOTest extends TrainingTests {

    @Test
    void from_createsTrainingDuration_onValidDays() {
        // given
        int validDays = 3;

        // when
        TrainingDurationVO duration = TrainingDurationVO.from(validDays);

        // then
        assertThat(duration).isNotNull();
        assertThat(duration.getDays()).isEqualTo(validDays);
    }

    @Test
    void from_throwsException_onInvalidDays() {
        //TODO: implement test for number of days validation
    }

    @Test
    void empty_createsTrainingDuration_withZeroDays() {
        // when
        TrainingDurationVO duration = TrainingDurationVO.empty();

        // then
        assertThat(duration).isNotNull();
        assertThat(duration.getDays()).isZero();
    }

} 

File: src/Ddd.Tests/Trainings/Idea/Domain/TrainingDurationVOTest.cs
using Ddd.Tests.Trainings.Tests;
using Ddd.Trainings.Idea.Domain;
using Xunit;

namespace Ddd.Tests.Trainings.Idea.Domain.Tests;

public class TrainingDurationVOTest : TrainingTests
{
  [Fact]
  public void From_CreatesTrainingDuration_OnValidDays()
  {
    // given
    int validDays = 3;

    // when
    TrainingDurationVO duration = TrainingDurationVO.From(validDays);

    // then
    Assert.NotNull(duration);
    Assert.Equal(validDays, duration.Days);
  }

  [Fact]
  public void From_ThrowsException_OnInvalidDays()
  {
    //TODO: implement test for number of days validation
  }

  [Fact]
  public void Empty_CreatesTrainingDuration_WithZeroDays()
  {
    // when
    TrainingDurationVO duration = TrainingDurationVO.Empty();

    // then
    Assert.NotNull(duration);
    Assert.Equal(0, duration.Days);
  }
}